flow-server/src/main/java/com/vaadin/flow/dom/impl/StyleAttributeHandler.java

```diff
@@ -65,6 +59,10 @@ public void setAttribute(Element element, String attributeValue) {
         parseStyles(attributeValue).forEach(style::set);
     }
 
+    private static final char SEMICOLON = ";".charAt(0);
+    private static final char PARENTHESIS_OPEN = "(".charAt(0);
+    private static final char PARENTHESIS_CLOSED = "(".charAt(0);
```

Comment:
```suggestion
    private static final char PARENTHESIS_CLOSED = ')';
```
This doesn't seem right at all and any test testing an open close should probably fail. 
`parseStyles` method requires tests for its parsing functionality.

Reply:
Please elaborate. You think the "parser" should detect non-closed functions? I changed this to this kind of implementation mostly to support semicolons in the URL values. As this is so little used "feature", I'm not convinved  it helps anybody, but that would be easy to add.

Reply:
Well a parenthesis is `(` to `)` but at the moment the check checks from `(` to `(` so for instance `name:(hello);fame:(oh no)" would not see any valueTerminated for hello.
Thus the method itself would need some validating tests with input->output

Reply:
Eh 🤦‍♂️ Looks like complex tests don't include setting multiple css rules at all 🤣 

The tests even worked with the dyslexia issue (once completed a test to fail) as the rest of the rule was considered as value. Thus addedd couple of cases where it is tested via the Style object that rules are propertly split.

An easy enhancement/validation could be to throw an error if a css function is not closed, but I'm not sure if it pays of (the expectation is that the feature is not really used and will be used even less in the future).